src/input.c

```diff
@@ -610,7 +606,8 @@ void _glfwPollAllJoysticks()
     {
         if (_glfw.joysticks[jid].connected == GLFW_TRUE)
         {
-            _glfwPlatformPollJoystick(_glfw.joysticks + jid, _GLFW_POLL_ALL);
+            _glfw.platform.pollJoystick(_glfw.joysticks + jid, _GLFW_POLL_ALL);
+            glfwPostEmptyEvent();
```

Comment:
post this empty event to force the joystick event out of the queue for whatever reason.